core/controllers/contributor_dashboard.py

```diff
@@ -260,6 +263,73 @@ def _content_in_review(self, state_name, content_id, suggestions):
             s.change.content_id == content_id for s in suggestions)
 
 
+class MachineTranslatedStateTextsHandler(base.BaseHandler):
+    """Provide a machine translation of exploration content."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.open_access
+    def get(self):
+        """Handles GET requests. Responds with a mapping from content id to
+        translation of form:
+            dict('translated_texts', dict(str, str|None))
+        If no translation is found for a given content id, that id is mapped to
+        None.
+        """
+        exp_id = self.request.get('exp_id')
+        if not exp_id:
+            raise self.InvalidInputException('Missing exp_id')
+
+        state_name = self.request.get('state_name')
+        if not state_name:
+            raise self.InvalidInputException('Missing state_name')
+
+        content_ids_string = self.request.get('content_ids')
+        content_ids = None
+        try:
+            content_ids = json.loads(content_ids_string)
+        except:
+            raise self.InvalidInputException(
+                'Improperly formatted content_ids %s' % content_ids_string)
+
+        target_language_code = self.request.get('target_language_code')
+        if not target_language_code:
+            raise self.InvalidInputException('Missing target_language_code')
+
+        # TODO(#12341): Tidy up this logic once we have a canonical list of
+        # language codes.
+        if not utils.is_supported_audio_language_code(
+                target_language_code
+            ) and not utils.is_valid_language_code(
+                target_language_code
+            ):
+            raise self.InvalidInputException(
+                'Invalid target_language_code: %s' % target_language_code)
+
+        exp = exp_fetchers.get_exploration_by_id(exp_id, strict=False)
+        if exp is None:
+            raise self.PageNotFoundException('Invalid exp_id: %s' % exp_id)
+        state_names_to_content_id_mapping = exp.get_translatable_text(
+            target_language_code)
+        if not state_names_to_content_id_mapping.has_key(state_name):
```

Comment:
Here and elsewhere: I think has_key is deprecated in Python 3. Just use "in".

Ditto below/elsewhere.

Reply:
Fixed.